src/magicbox/SuggestionsManager.ts

```diff
@@ -306,29 +336,85 @@ export class SuggestionsManager {
     return $$(dom);
   }
 
-  private move(direction: 'up' | 'down') {
+  private move(direction: direction) {
+    const previewSelectables = $$(this.element).findAll(`.coveo-preview-selectable`);
+    if (previewSelectables.length > 0) {
+      return this.moveWithQuerySuggestPreview(direction);
+    }
+    return this.moveWithinSuggestion(direction);
+  }
+
+  private moveWithinSuggestion(direction: direction) {
     const currentlySelected = $$(this.element).find(`.${this.options.selectedClass}`);
     const selectables = $$(this.element).findAll(`.${this.options.selectableClass}`);
     const currentIndex = indexOf(selectables, currentlySelected);
 
     let index = direction == 'up' ? currentIndex - 1 : currentIndex + 1;
-    if (index < -1) {
-      index = selectables.length - 1;
-    }
-    if (index > selectables.length) {
-      index = 0;
-    }
+    index = (index + selectables.length) % selectables.length;
 
-    const newlySelected = selectables[index];
+    this.lastSelectedSuggestion = selectables[index];
 
-    if (newlySelected) {
-      this.processKeyboardSelection(newlySelected);
+    return this.selectQuerySuggest(this.lastSelectedSuggestion);
+  }
+
+  private selectQuerySuggest(suggestion: HTMLElement) {
+    if (suggestion) {
+      this.processKeyboardSelection(suggestion);
     } else {
       this.keyboardFocusedSuggestion = null;
       this.inputManager.input.removeAttribute('aria-activedescendant');
     }
 
-    return newlySelected;
+    return suggestion;
+  }
+
+  private moveWithQuerySuggestPreview(direction: direction) {
+    const currentlySelected = $$(this.element).find(`.${this.options.selectedClass}`);
+    const omniboxSelectables = $$(this.element).findAll(`.${this.options.selectableClass}`);
+    const previewSelectables = $$(this.element).findAll(`.coveo-preview-selectable`);
+    const omniboxIndex = indexOf(omniboxSelectables, currentlySelected);
+    const previewIndex = indexOf(previewSelectables, currentlySelected);
+    const noSelection = omniboxIndex == -1 && previewIndex == -1;
+
+    //if we move within the suggestion
+    if ((noSelection || omniboxIndex > -1) && (direction == 'up' || direction == 'down')) {
+      return this.moveWithinSuggestion(direction);
+    }
+
+    //if there is no preview OR we go left when we are on the Suggestion
+    if (previewSelectables.length == 0 || (direction == 'left' && previewIndex == -1)) {
```

Comment:
Similar comment as above of breaking up the condition into well-named variables and removing the need for a comment.